plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/utils/PlayAssetDeliveryUtils.java

```diff
@@ -130,6 +131,22 @@ public static Dictionary convertAssetPackLocationsToDictionary(
             (d1, d2) -> d1.putAll(d2));
   }
 
+  /**
+   * Serializes an Exception object into Godot Dictionary. If the Exception is an
+   * AssetPackException, the errorCode entry in returnDict will be populated.
+   *
+   * @param e Exception to be converted to Dictionary
+   * @return serialized Dictionary
+   */
+  public static Dictionary convertExceptionToDictionary(Exception e) {
+    Dictionary returnDict = new Dictionary();
+    returnDict.put("toString", e.toString());
+    if (e instanceof AssetPackException) {
+      returnDict.put("errorCode", ((AssetPackException) e).getErrorCode());
```

Comment:
Since this exception will rarely be anything other than AssetPackException, it might be good to always put an `errorCode` in the dict, and default it to [`INTERNAL_ERROR`](https://developer.android.com/reference/native/play/core/group/assetpack#group__assetpack_1ggab72339a7d36069c9bcd9dbc4059bb03ea1f984b7c4c84eab758ebd49e6bb82ac1) if the Exception is not an AssetPackException.